addons/mail/static/tests/activity/activity_tests.js

```diff
@@ -0,0 +1,44 @@
+/** @odoo-module **/
+
+import { Activity } from "@mail/activity/activity";
+import { activityData } from "@mail/../tests/activity/activityData";
+import { click, getFixture, mount } from "@web/../tests/helpers/utils";
+import { makeTestEnv, TestServer } from "../helpers/helpers";
+
+let target;
+
+QUnit.module("mail", (hooks) => {
+    hooks.beforeEach(async () => {
+        target = getFixture();
+    });
+
+    QUnit.module("activity");
+
+    QUnit.test("activity", async (assert) => {
+        const server = new TestServer();
+        const env = makeTestEnv((route, params) => {
+            if (route === "/web/dataset/call_kw/mail.activity/unlink") {
+                assert.step("/web/dataset/call_kw/mail.activity/unlink");
+            }
+            return server.rpc(route, params);
+        });
+        await mount(Activity, target, {
+            env,
+            props: { data: activityData },
+        });
+        assert.equal(
+            document.querySelector(".o-mail-activity-name").textContent,
+            activityData.activity_type_id[1]
+        );
+        assert.equal(
+            document.querySelector(".o-mail-activity-user").textContent,
+            "for " + activityData.user_id[1]
+        );
+        await click(document.querySelector(".o-mail-activity-toggle"));
+        assert.containsOnce(target, ".o-mail-activity-details");
+        await click(document.querySelector(".o-mail-activity-toggle"));
+        assert.containsNone(target, ".o-mail-activity-details");
+        await click(document.querySelector(".o-mail-activity-unlink-button"));
+        assert.verifySteps(["/web/dataset/call_kw/mail.activity/unlink"]);
+    });
```

Comment:
The test should be named in a way that makes it clear what is being tested.
You might want to split this test in several properly named tests.